WordPress/gutenberg
# Add Reverse on Mobile style for Columns and Media & Text Blocks

## What?
Fixes #40706 and #26215

Adds a style `Reverse on mobile` to the Columns and Media & Text Blocks. When selected, the block's column order will be reversed if the screen is narrower than the block's mobile breakpoint.

For Columns Blocks, this means the column order will simply be reverse. For Media & Text Blocks, this means that the content will appear on top and the media will appear on bottom, because default behavior is for media to appear on top and content to appear on bottom for mobile screen widths.

## Why?

Columns Blocks and Media & Text Blocks are frequently used to display media and content in a zig-zag pattern. For example:
```
[image]   [text ]
[text ]   [image]
[image]   [text ]
```

For this type of pattern, users typically want the content to stack on mobile like this:

```
[image]
[text ]
[image]
[text ]
[image]
[text ]
```

With the current Columns Block behavior, the content ends up looking like this:

```
[image]
[text ]
[text ]
[image]
[image]
[text ]
```

## How?

The Reverse on mobile style allows users greater flexibility in determining how their content is translated from unstacked to stacked. Users can choose the order in which the stacked content will appear.

## Testing Instructions

1. Open a post or page.
2. Add at least 3 or 4 Columns Blocks, with 2 columns each.
3. Add media and content in a zig-zag pattern, as described above.
4. Apply the "Reverse on mobile" style to every other Columns Block.
5. Observe the stacking behavior in the editor both with and without the style applied.
6. Repeat the previous steps with the Media & Text blocks, except apply the "Reverse on mobile" style to all blocks.

### Testing Instructions for Keyboard

1. Insert a Columns or Media & Text Block using existing methods.
2. Open the Settings sidebar using existing methods.
3. Use the Tab key to highlight Settings, then use the right arrow to change the selection to Styles.
4. Use the Tab key to highlight Reverse on mobile, and use Enter to select the style.

## Screenshots or screencast 
<img width="1672" alt="Screenshot 2023-10-31 at 10 09 35 PM" src="https://github.com/WordPress/gutenberg/assets/12906034/32b97343-7662-4673-b854-9107fc8a9148">

Screenshot showing the location of the 'Reverse on mobile' style in the editor, along with Columns blocks displaying the aforementioned 'zig-zag' pattern.

<img width="999" alt="Screenshot 2023-10-31 at 10 11 02 PM" src="https://github.com/WordPress/gutenberg/assets/12906034/8093f7a0-155f-4af5-9f1d-e33535cce505">

Screenshot showing the appearance of stacked Columns blocks in the editor. The 'Reverse on mobile' style has been applied to every other Columns Block so that the 'zig-zag' pattern translates to an appropriate stacked pattern.

<img width="700" alt="Screenshot 2023-10-31 at 10 11 15 PM" src="https://github.com/WordPress/gutenberg/assets/12906034/3f48d6d8-bbbf-471c-8615-4437b3dd6fc0">

Screenshot showing the front end of a page with stacked Columns Blocks, with every other Columns Block having the 'Reverse on mobile' style.
